book/analysis/big-o-notation.md

```diff
@@ -5,104 +5,106 @@ position: 2
-
-In our summation algorithms shown earlier, a good basic unit of
-computation  might be to count the number of assignment statements
-performed to compute the sum. In the function `sum_of_n`, the number of
-assignment statements is 1 (`the_sum = 0`) plus the value of *n* (the
-number of times we perform `the_sum=the_sum+i`). We can denote this by
-a function, call it T, where $$T(n)=1 + n$$. The parameter *n* is often
-referred to as the “size of the problem,” and we can read this as “T(n)
-is the time it takes to solve a problem of size n, namely 1 + n steps.”
-
-In the summation functions given earlier, it makes sense to use the number
-of terms in the summation to denote the size of the problem. We can then
-say that the sum of the first 100,000 integers is a bigger instance of
-the summation problem than the sum of the first 1,000. Because of this,
-it might seem reasonable that the time required to solve the larger case
-would be greater than for the smaller case. Our goal then is to show how
-the algorithm’s execution time changes with respect to the size of the
-problem.
-
-Computer scientists prefer to take this analysis technique one step
-further. It turns out that the exact number of operations is not as
-important as determining the most dominant part of the $$T(n)$$ function.
-In other words, as the problem gets larger, some portion of the $$T(n)$$
-function tends to overpower the rest. This dominant term is what, in the
-end, is used for comparison. The *order of magnitude* function
-describes the part of $$T(n)$$ that increases the fastest as the value of
-*n* increases. Order of magnitude is often called *big O* notation
-(for “order”) and written as $$O(f(n))$$. It provides a useful
-approximation to the actual number of steps in the computation. The
-function $$f(n)$$ provides a simple representation of the dominant part of
-the original $$T(n)$$.
-
-In the above example, $$T(n)=1+n$$. As *n* gets large, the constant 1 will
-become less and less significant to the final result. If we are looking
-for an approximation for $$T(n)$$, then we can drop the 1 and simply say
-that the running time is $$O(n)$$. It is important to note that the 1 is
-certainly significant for $$T(n)$$. However, as *n* gets large, our
-approximation will be just as accurate without it.
+An algorithm is little more than a series of steps required to perform some
+task. If we treat each step as a basic unit of computation, then an algorithm's
+execution time can be expressed as the *number of steps required to solve the
+problem*.
+
+This abstraction is exactly what we need: it characterizes an algorithm’s
+efficiency in terms of execution time while remaining independent of any
+particular program or computer.
+
+How can we apply this abstraction to the summation algorithms shown earlier?
+
+Looking back at our function `sum_of_n`, the most expensive unit of computation
+seems to be variable assignment so, if we count those, we could have a worthy
```

Comment:
@aleksharma12 you're the pro, but seems like it would flow better as "... seems to be variable assignment, so if we could those we could have a worthy..."